lms/djangoapps/courseware/module_render.py

```diff
@@ -1002,11 +1008,34 @@ def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
         suffix (str)
 
     Raises:
+        HttpResponseForbidden: If the request method is not in SAFE_METHODS and user is not authenticated.
         Http404: If the course is not found in the modulestore.
     """
-    # NOTE (CCB): Allow anonymous GET calls (e.g. for transcripts). Modifying this view is simpler than updating
-    # the XBlocks to use `handle_xblock_callback_noauth`...which is practically identical to this view.
-    if request.method != 'GET' and not request.user.is_authenticated:
+    # In this case, we are using Session based authentication, so we need to check CSRF token.
+    if request.user.is_authenticated:
+        error = CsrfViewMiddleware().process_view(request, None, (), {})
+        if error:
+            return error
+
+    else:
+        authentication_classes = (JwtAuthentication, OAuth2Authentication)
```

Comment:
Could you please add a comment to the code, explaining why we are using this custom authentication handling, instead of just using the normal DRF decorators? Otherwise people reading the code in the future, without this PR for context, may refactor and reinstate the bug this is trying to solve.

Reply:
That's a good point, @bradenmacdonald!
I added the relevant comment.